File: Sample.Products.Backend/Sample.Products.Backend.Entities/Concrete/Tables/Product.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;
using Sample.Products.Backend.Entities.Abstract;

namespace Sample.Products.Backend.Entities.Concrete.Tables
{
    public class Product:BaseEntity
    {
        public string Name { get; set; }
        public string ShortDescription { get; set; }
        public string Description { get; set; }
        public int Quantity { get; set; }
        public decimal Price { get; set; }
        public int BrandId { get; set; }

        [JsonIgnore]
        public virtual Brand Brand { get; set; }
        [JsonIgnore]
        public virtual ICollection<ProductPicture> ProductPictures { get; set; }
        [JsonIgnore]
        public virtual ICollection<ProductCategory> ProductCategories { get; set; }
        [JsonIgnore]
        public virtual ICollection<ProductTag> ProductTags { get; set; }
        // public virtual ICollection<Tag> Tags { get; set; }
        // public virtual ICollection<Picture> Pictures { get; set; }
        // public virtual ICollection<Category> Categories { get; set; }
        
    }
}

File: sample_products_mobile_app/lib/utils/models/product.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:sample_products_mobile_app/utils/models/brand.dart';
import 'package:sample_products_mobile_app/utils/models/tag.dart';

part 'product.g.dart';

@JsonSerializable()
class Product {
  late int? id;
  late String? name;
  late String? shortDescription;
  late String? description;
  late int? quantity;
  late double? price;
  late int? brandId;

  late Brand? brand;
  late List<Tag>? tags;
  // late  ICollection<PictureModel> Pictures ;
  late List<int>? categoryIds;
  late List<int>? pictureIds;

  Product(
      {this.id,
      this.name,
      this.brand,
      this.brandId,
      this.categoryIds,
      this.description,
      this.pictureIds,
      this.price,
      this.quantity,
      this.shortDescription,
      this.tags});

  factory Product.fromJson(Map<String, dynamic> json) =>
      _$ProductFromJson(json);
  Map<String, dynamic> toJson() => _$ProductToJson(this);
}
